RadarSDK/RadarAPIHelper.m

```diff
@@ -67,9 +67,38 @@ - (void)requestWithMethod:(NSString *)method
             }
 
             if (params) {
-                [req setHTTPBody:[NSJSONSerialization dataWithJSONObject:params options:0 error:NULL]];
+                if (!params[@"updatedAtMsDiff"] && !params[@"replays"]) {
+                    [req setHTTPBody:[NSJSONSerialization dataWithJSONObject:params options:0 error:NULL]];
+                } else {
+                    NSMutableDictionary *requestParams = [params mutableCopy];
+                    NSNumber *locationMs = params[@"locationMs"];
+                    long nowMs = (long)([NSDate date].timeIntervalSince1970 * 1000);
+
+                    if (locationMs) {
+                        long updatedAtMsDiff = nowMs - [locationMs longValue];
+                        requestParams[@"updatedAtMsDiff"] = @(updatedAtMsDiff);
+                    }
+
+                    NSArray *replays = params[@"replays"];
+                    if (replays) {
+                        NSMutableArray *updatedReplays = [NSMutableArray arrayWithCapacity:replays.count];
+                        for (NSDictionary *replay in replays) {
+                            NSMutableDictionary *updatedReplay = [replay mutableCopy];
+                            NSNumber *replayLocationMs = replay[@"locationMs"];
+                            if (replayLocationMs) {
+                                long replayUpdatedAtMsDiff = nowMs - [replayLocationMs longValue];
+                                updatedReplay[@"updatedAtMsDiff"] = @(replayUpdatedAtMsDiff);
+                            }
+                            [updatedReplays addObject:updatedReplay];
+                        }
+                        requestParams[@"replays"] = updatedReplays;
+                    }
```

Comment:
we should put this logic in `RadarAPIClient flushReplays:completionHandler:`

Reply:
Yea, think that's a more natural spot for it, but also, I think we want to do the top level `updatedAtMsDiff` replacement in `APIHelper` because if `/track` is called twice, there will be a sleep interval, and we want to replace it after that time.

So the option would be to have the above in `RadarAPIHelper` and the replay replacement in `RadarAPIClient flushReplays:completionHandler` or have it all in the same place in `RadarAPIHelper`.

I'm not sure, but I lean towards the latter I think.